plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -417,6 +417,13 @@ public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoCol
         return iterable.iterator();
     }
 
+    static Document buildFilter(MongoTableHandle table)
+    {
+        Document filter = buildQuery(table.getConstraint());
+        table.getFilter().ifPresent(filter::putAll);
```

Comment:
nit: Can we change the order, first we build a filter that is provided in PTF and then we could push the filter from the query.